lib/commands/new.js

```diff
@@ -29,6 +30,27 @@ module.exports = Command.extend({
     '<app-name>'
   ],
 
+  beforeRun: function(rawArgs) {
+    if (!rawArgs.length) {
+      return;
+    }
+    // merge in blueprint availableOptions
+    var blueprint;
+    try{
+      blueprint = this.lookupBlueprint(rawArgs[0]);
+      this.registerOptions(blueprint);
+    }
+    catch(e) {
+      SilentError.debugOrThrow('ember-cli/commands/init', e);
```

Comment:
@bendemboski yes, a single module in a method sounds great.

Another approach could be:

```js
Command.extend({
  mergeBlueprintOptions: mergeBlueprintOptions,
  beforeRun: function(rawArgs) {
    this.mergeBlueprintOptions(rawArgs);
  }
})
```
or even more concisely 

```js
Command.extend({
  beforeRun: mergeBlueprintOptions
})
```